### PR TITLE
Automate bindings: enable binding of abstract classes

### PR DESCRIPTION
**Pull request by zer0main**
No reason to disable them, just bind without make function and creation tests.

### CODE REVIEW

File: tools/automate_bindings.py [hunk 1]
```diff
@@ -639,32 +639,40 @@ def generateMethodsArray(module_name, methods):
...
-    constructor_name = 'make'
-    constructor_return_type = module_name + ' *'
-    return implementLuaCFunction(
-        True,
-        module_name,
-        constructor_name,
-        [c.arguments for c in constructors],
-        constructor_return_type,
-    )
+    if len(constructors) != 0:
```
Comment (starius) [inline tools/automate_bindings.py]: 1. Use [Guard Clause](http://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html) here.
2. You can write just `if constructors:`

File: tools/automate_bindings.py [hunk 2]
```diff
@@ -840,7 +848,9 @@ def bind(modules, module_only, blacklist):
             )
             if not module_only:
                 addModuleToLists(module_name, global_namespace.namespace('Wt'))
-            addTest(module_name)
+            # Is not abstract
+            if len(constructors):
```
Comment (starius) [inline tools/automate_bindings.py]: `if constructors:`

File: tools/automate_bindings.py [hunk 3]
```diff
@@ -705,7 +713,7 @@ def generateModule(module_name, methods, base, constructors, signals):
         ))
     source.append(generateSignals(signals, module_name))
     source.append(generateMethodsArray(module_name, methods + signals))
-    source.append(generateModuleFunc(module_name, base))
+    source.append(generateModuleFunc(module_name, base, len(constructors)))
```
Comment (starius) [inline tools/automate_bindings.py]: `source.append(generateModuleFunc(module_name, base, is_not_abstract=bool(constructors)))`

File: tools/automate_bindings.py [hunk 4]
```diff
@@ -284,14 +282,16 @@ def getMembers(global_namespace, module_name, blacklisted):
 def getConstructors(global_namespace, module_name, blacklisted):
     Wt = global_namespace.namespace('Wt')
     main_class = Wt.class_(name=module_name)
+    result = []
+    if main_class.is_abstract:
+        return result
```
Comment (starius) [inline tools/automate_bindings.py]: `return []` and declare `result` below, as before.

File: tools/automate_bindings.py [hunk 5]
```diff
@@ -840,7 +848,9 @@ def bind(modules, module_only, blacklist):
             )
             if not module_only:
                 addModuleToLists(module_name, global_namespace.namespace('Wt'))
-            addTest(module_name)
+            # Is not abstract
```
Comment (starius) [inline tools/automate_bindings.py]: Move this comment inside the block under `if`.

**Review by starius — CHANGES_REQUESTED** [review]

File: tools/automate_bindings.py (hunk 5 above)
Comment (zer0main) [reply tools/automate_bindings.py]: Done :+1:

File: tools/automate_bindings.py (hunk 2 above)
Comment (zer0main) [reply tools/automate_bindings.py]: Done :+1:

File: tools/automate_bindings.py (hunk 3 above)
Comment (zer0main) [reply tools/automate_bindings.py]: Done :+1:

File: tools/automate_bindings.py (hunk 4 above)
Comment (zer0main) [reply tools/automate_bindings.py]: Done :+1:

File: tools/automate_bindings.py (hunk 1 above)
Comment (zer0main) [reply tools/automate_bindings.py]: Done :+1:

**Review by starius — APPROVED** [review]

**Pull request merged**